pkg/process/checks/interval.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	ProcessCheckDefaultInterval          = 10 * time.Second
+	RTProcessCheckDefaultInterval        = 2 * time.Second
+	ContainerCheckDefaultInterval        = 10 * time.Second
+	RTContainerCheckDefaultInterval      = 2 * time.Second
+	ConnectionsCheckDefaultInterval      = 30 * time.Second
+	PodCheckDefaultInterval              = 10 * time.Second
+	ProcessDiscoveryCheckDefaultInterval = 4 * time.Hour
+
+	discoveryMinInterval = 10 * time.Minute
+
+	configIntervals = configPrefix + ".intervals"
+
+	// The interval, in seconds, at which we will run each check. If you want consistent
+	// behavior between real-time you may set the Container/ProcessRT intervals to 10.
+	// Defaults to 10s for normal checks and 2s for others.
+	configProcessInterval     = configIntervals + ".process"
+	configRTProcessInterval   = configIntervals + ".process_realtime"
+	configContainerInterval   = configIntervals + ".container"
+	configRTContainerInterval = configIntervals + ".container_realtime"
+	configConnectionsInterval = configIntervals + ".connections"
+)
+
+var (
+	defaultIntervals = map[string]time.Duration{
+		ProcessCheckName:       ProcessCheckDefaultInterval,
+		RTProcessCheckName:     RTProcessCheckDefaultInterval,
+		ContainerCheckName:     ContainerCheckDefaultInterval,
+		RTContainerCheckName:   RTContainerCheckDefaultInterval,
+		ConnectionsCheckName:   ConnectionsCheckDefaultInterval,
+		PodCheckName:           PodCheckDefaultInterval,
+		DiscoveryCheckName:     ProcessDiscoveryCheckDefaultInterval,
+		ProcessEventsCheckName: config.DefaultProcessEventsCheckInterval,
```

Comment:
Why did you decide to keep the process events check interval in `pkg/config`?

Reply:
I believe it was put there originally by @mbotarro, we can move all the defaults there eventually.

Reply:
IMO we should be decisive while we're moving stuff around, since it's unlikely that we're going to go back and fix it otherwise. However I'll leave it up to you to decide whether or not you want to leave it.

Reply:
To explain my thinking here, so it's a bit clearer why I am not doing it in this PR:
- This is where we have consts defined for the new events check: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/process.go#L62-L69
- The default value (`DefaultProcessEventsCheckInterval`) is indeed covered in a test in `pkg/config` here: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/process_test.go#L127
- The min value is not covered by tests in `pkg/config`

My primary reason for not moving these consts in/out of `pkg/config` is that it may create more confusion here and it's easier to do it separately. Since min values aren't currently enforced there, it's probably best to keep them here (where they are used), defaults are, but we are also using them here in some case.

What I would do:
- Keep defaults only used in `pkg/config` unexported.
- Keep min values in `pkg/check` where they are enforced.

Let me know if this makes sense.

Reply:
That makes sense to me! Let's keep it as it is for now